### PR TITLE
[database] new, API table metadata

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [X] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Migrate `/table/<database_id>/<table_name>/<schema>/` to `/api/v1/database/<database_id>/table/<table_name>/<schema_name>/` resource.

- Guarantees database filtering based on the user permissions
- Full OpenAPI spec
- Better error handling

Response spec from swagger UI:
image

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [X] Introduces new feature or API
- [X] Removes existing feature or API

### CODE REVIEW

File: superset/views/database/api.py [hunk 1]
```diff
@@ -14,20 +14,88 @@
...
+from superset import event_logger
+from superset.models.core import Database
+from superset.utils.core import error_msg_from_exception, parse_js_uri_path_item
 from superset.views.base_api import BaseSupersetModelRestApi
+from superset.views.database.filters import DatabaseFilter
+from superset.views.database.mixins import DatabaseMixin
+from superset.views.database.validators import sqlalchemy_uri_validator
+
+
+def get_table_schema_info(
```
Comment (dpgaspar) [inline superset/views/database/api.py]: Not very happy with placing this here...

**Review by craig-rueda — APPROVED** [review]
LGTM

File: superset/views/database/api.py [hunk 2]
```diff
@@ -50,3 +137,144 @@ class DatabaseRestApi(DatabaseMixin, BaseSupersetModelRestApi):
...
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        table_name_parsed: Optional[str] = parse_js_uri_path_item(table_name)
+        schema_parsed: Optional[str] = parse_js_uri_path_item(
+            schema_name, eval_undefined=True
+        )
```
Comment (villebro) [inline superset/views/database/api.py]: As the return type is already typed in `parse_js_uri_path_item`, do these need to be explicitly typed here?

**Review by villebro — APPROVED** [review]
One small nit/question, other than that LGTM

File: superset/views/database/api.py (hunk 2 above)
Comment (dpgaspar) [reply superset/views/database/api.py]: yes, good question. Not really I guess, what do you think?

Comment (villebro) [reply superset/views/database/api.py]: I think it's already implied i.e.not needed, so would personally remove it unless it makes the code more understandable.

Comment (dpgaspar) [reply superset/views/database/api.py]: removed

**Pull request merged**